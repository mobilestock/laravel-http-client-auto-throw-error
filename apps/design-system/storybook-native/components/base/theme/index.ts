import { Dimensions, PixelRatio, Platform, StatusBar } from 'react-native'

export const theme = {
  cores: {
    corSecundaria: '#4ba8ff',
    botaoPadrao: '#4B6680',
    backgroundGray: '#7D8D9C',
    tomate: '#FF6C63',

    azul100: '#4ba8ff',
    azul40: '#222465',
    azul30: '#014C76',
    azul20: '#96CCFF',
    azul: 'blue',

    branco90: '#f1f1f1',
    branco: '#fff',

    bege10: '#c4c4c4',

    preto: '#000',

    texto: '#292929',

    sucesso: '#0cb536',

    alerta80: '#F51D07',
    alerta80A_bb: '#F51D07bb',
    alerta70: '#FF3D0D',
    alerta50: '#E86513',
    alerta20: '#F2BF2C',

    whatsapp: '#128c7e',

    situacoes: {
      conferencia: {
        background: '#363636',
        color: '#eeeeee'
      },
      expedicao: {
        background: '#008DDC',
        color: '#f1f1f1'
      },
      pontoTransporte: {
        background: '#DC7700',
        color: '#f1f1f1'
      },
      entregarCliente: {
        background: '#00DC58',
        color: '#000'
      },
      entregue: {
        background: '#00DC58',
        color: '#f1f1f1'
      },
      devolucao: {
        background: '#8B71D9',
        color: '#000'
      },
      monitoramento: {
        color: '#FFF',
        background: '#4ba8ff'
      }
    }
  },
  fonts: {
    size(fontSize: number, standardScreenHeight = 680): number {
      const { height, width } = Dimensions.get('window')
      const standardLength = width > height ? width : height
      let offset = 0
      let heightPercent = 0

      if (width > height) {
        offset = 0
      } else if (Platform.OS === 'ios') {
        offset = 190
      } else {
        offset = StatusBar.currentHeight || 0
      }

      const deviceHeight = Platform.OS !== 'android' ? standardLength - offset : standardLength

      if (typeof fontSize === 'number') {
        heightPercent = (fontSize * deviceHeight) / standardScreenHeight
      } else {
        heightPercent = (parseFloat(fontSize) * deviceHeight) / 100
      }
      return Math.round(heightPercent)
    }
  },
  layout: {
    size: (size: number): number => {
      if (PixelRatio.get() < 2) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 1.3
      }
      if (PixelRatio.get() >= 2 && PixelRatio.get() < 3) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 0.9
      }
      if (PixelRatio.get() >= 3) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 0.7
      }
      return 16
    },
    width: (widthPercent: number): number => {
      let screenWidth = Dimensions.get('window').width
      const elemWidth = typeof widthPercent === 'number' ? widthPercent : parseFloat(widthPercent)
      return PixelRatio.roundToNearestPixel((screenWidth * elemWidth) / 100)
    },
    height: (heightPercent: number): number => {
      let screenHeight = Dimensions.get('window').height
      const elemHeight = typeof heightPercent === 'number' ? heightPercent : parseFloat(heightPercent)
      return PixelRatio.roundToNearestPixel((screenHeight * elemHeight) / 100)
    }
  }
}
