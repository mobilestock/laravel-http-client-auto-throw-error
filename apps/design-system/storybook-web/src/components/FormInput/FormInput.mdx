import { Canvas, Controls, Meta, Story } from '@storybook/blocks';
import * as FormInputStories from './index.stories.tsx';

<Meta of={FormInputStories} />

# FormInput

O componente **FormInput** é utilizado para criar campos de entrada em formulários. Ele oferece suporte a diferentes tipos de entrada, como texto, telefone, e senha, e pode ser configurado para exibir mensagens de erro e ícones de visibilidade para campos de senha.

## Uso Básico

Você pode utilizar o componente `FormInput` para criar um campo de entrada simples com suporte a validação e formatação. Aqui está um exemplo básico de uso:

<Canvas of={FormInputStories.Default} />
<Controls />

## Propriedades

- **name**: Nome do campo utilizado pelo Unform para identificação.
- **placeholder**: Placeholder exibido dentro do campo de entrada.
- **type**: Tipo do campo de entrada (`text`, `tel`, `email`, `password`).
- **showErrorMessage**: Exibe mensagem de erro abaixo do campo, se houver.
- **label**: Rótulo exibido acima do campo de entrada.
- **format**: Função opcional para formatar o valor do campo.

## Exemplos de Uso

### Campo de Telefone com Mensagem de Erro

Use a prop `showErrorMessage` para exibir mensagens de erro abaixo do campo de entrada. Isso é útil para validar o valor do campo e informar o usuário.

<Canvas of={FormInputStories.WithErrorMessage} />
