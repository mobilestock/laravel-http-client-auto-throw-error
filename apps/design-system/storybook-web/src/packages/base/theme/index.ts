export const theme = {
  colors: {
    background: '#f05'
  }
}
