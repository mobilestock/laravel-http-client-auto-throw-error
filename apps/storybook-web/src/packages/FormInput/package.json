{
  "name": "@mobilestock/form_input",
  "version": "0.0.0",
  "main": "./src/index.tsx",
  "files": [
    "*"
  ],
  "exports": {
    ".": {
      "default": "./src/index.tsx"
    }
  },
  "dependencies": {
    "@mobilestock/input": "^0.0.0"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "styled-components": "^6.1.8"
  }
}