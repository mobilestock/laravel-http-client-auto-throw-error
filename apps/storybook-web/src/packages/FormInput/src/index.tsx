import { useEffect, useRef } from 'react'

import { useField } from '@unform/core'

import Input, { InputProps } from '@mobilestock/input'

interface FormInputProps extends InputProps {
  name: string
}

export default function FormInput(props: FormInputProps) {
  const { fieldName, defaultValue, registerField, error } = useField(props.name)
  const inputRef = useRef<HTMLInputElement>(null)

  useEffect(() => {
    registerField({
      name: fieldName,
      ref: inputRef,
      getValue: ref => ref.current.value,
      setValue: (ref, value) => {
        ref.current.value = value
      },
      clearValue: ref => {
        ref.current.value = ''
      }
    })
  }, [fieldName, registerField])

  return <Input ref={inputRef} defaultValue={defaultValue} error={error} {...props} />
}
