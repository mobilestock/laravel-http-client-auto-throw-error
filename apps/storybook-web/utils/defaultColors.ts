export default {
  primary: '#000000',
  secondary: '#FFFFFF',
  tertiary: '#FF0000',

  blue41: '#303f9f',
  blue50: '#007FFF',
  shadow: 'rgba(0 0 0 0.25)',
  purpleShadow: 'rgba(101, 13, 73, 0.25)',
  grey75: '#C0C0C0C0',
  grey7: '#121212',
  orange50: '#ffc700',
  yellow50: '#FFFF00',
}
