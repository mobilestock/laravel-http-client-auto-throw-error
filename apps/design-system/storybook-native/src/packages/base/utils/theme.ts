import { Dimensions, PixelRatio, Platform, StatusBar } from 'react-native'

import defaultColors from './defaultColors'

export const theme = {
  colors: {
    text: {
      primary: defaultColors.black,
      secondary: defaultColors.white,
    },

    container: {
      shadow: defaultColors.shadow,
      pure: defaultColors.white,
    },

    button: {
      default: defaultColors.primary,
    },
  },

  layout: {
    size: (size: number): number => {
      if (PixelRatio.get() < 2) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 1.3
      }
      if (PixelRatio.get() >= 2 && PixelRatio.get() < 3) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 0.9
      }
      if (PixelRatio.get() >= 3) {
        return PixelRatio.getPixelSizeForLayoutSize(size) * 0.7
      }
      return 16
    },
    width: (widthPercent: number): number => {
      let screenWidth = Dimensions.get('window').width
      const elemWidth = typeof widthPercent === 'number' ? widthPercent : parseFloat(widthPercent)
      return PixelRatio.roundToNearestPixel((screenWidth * elemWidth) / 100)
    },
    height: (heightPercent: number): number => {
      let screenHeight = Dimensions.get('window').height
      const elemHeight = typeof heightPercent === 'number' ? heightPercent : parseFloat(heightPercent)
      return PixelRatio.roundToNearestPixel((screenHeight * elemHeight) / 100)
    },
  },

  fonts: {
    size(fontSize: number, standardScreenHeight = 680): number {
      const { height, width } = Dimensions.get('window')
      const standardLength = width > height ? width : height
      let offset = 0
      let heightPercent = 0

      if (width > height) {
        offset = 0
      } else if (Platform.OS === 'ios') {
        offset = 190
      } else {
        offset = StatusBar.currentHeight || 0
      }

      const deviceHeight = Platform.OS !== 'android' ? standardLength - offset : standardLength

      if (typeof fontSize === 'number') {
        heightPercent = (fontSize * deviceHeight) / standardScreenHeight
      } else {
        heightPercent = (parseFloat(fontSize) * deviceHeight) / 100
      }
      return Math.round(heightPercent)
    },
  },
}
