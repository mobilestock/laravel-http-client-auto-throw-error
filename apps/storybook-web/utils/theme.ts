import defaultColors from './defaultColors'

export const theme = {
  colors: {
    text: {
      default: defaultColors.secondary,
      regular: defaultColors.primary,
      notice: defaultColors.orange50,
      danger: defaultColors.tertiary,
    },
    container: {
      default: defaultColors.grey7,
      outline: {
        default: defaultColors.grey75,
        soft: defaultColors.secondary,
      },
      pure: defaultColors.secondary,
      shadow: defaultColors.shadow,
      purpleShadow: defaultColors.purpleShadow,
    },
    button: {
      default: defaultColors.blue41,
      anchor: defaultColors.blue50,
      shadow: defaultColors.shadow,
    },
    alert: {
      urgent: defaultColors.tertiary,
      tip: defaultColors.yellow50,
    },
  },
}
