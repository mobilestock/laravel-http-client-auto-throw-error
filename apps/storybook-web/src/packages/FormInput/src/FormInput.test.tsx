import { fireEvent, render, waitFor } from '@testing-library/react'
import { useField } from '@unform/core'
import FormInput from '.'
import { theme } from '../../../../utils/theme'

jest.mock('@unform/core', () => ({
  useField: jest.fn()
}))

describe('FormInput Component - Web', () => {
  const mockRegisterField = jest.fn()
  const mockUseField = (fieldName: string, defaultValue = '', error = '') => {
    (useField as jest.Mock).mockReturnValue({
      fieldName,
      defaultValue,
      registerField: mockRegisterField,
      error
    })
  }

  beforeEach(() => {
    jest.clearAllMocks();
    (window.HTMLFormElement.prototype.requestSubmit as jest.Mock) = jest.fn()
  })

  it('deve renderizar o componente sem erros', () => {
    mockUseField('email')
    render(global.app(<FormInput name="email" />))
  })

  it('deve renderizar o valor padrão corretamente', () => {
    mockUseField('email', 'test@example.com')
    const { getByDisplayValue } = render(global.app(<FormInput name="email" />))
    expect(getByDisplayValue('test@example.com')).toBeInTheDocument()
  })

  it('deve registrar o campo corretamente', () => {
    mockUseField('email')
    render(global.app(<FormInput name="email" />))
    expect(mockRegisterField).toHaveBeenCalledWith({
      name: 'email',
      ref: expect.any(Object),
      getValue: expect.any(Function),
      setValue: expect.any(Function),
      clearValue: expect.any(Function)
    })
  })

  it('deve exibir o erro quando houver', () => {
    mockUseField('email', '', 'Campo obrigatório')
    const { getByText } = render(global.app(<FormInput name="email" />))
    expect(getByText('Campo obrigatório')).toBeInTheDocument()
  })

  it('deve permitir a alteração do valor do input', () => {
    mockUseField('email')
    const { getByRole } = render(global.app(<FormInput name="email" />))
    const input = getByRole('textbox')
    fireEvent.change(input, { target: { value: 'novo valor' } })
    expect((input as HTMLInputElement).value).toBe('novo valor')
  })

  it('deve limpar o valor do input quando solicitado', () => {
    mockUseField('email', 'valor inicial')
    const { getByRole } = render(global.app(<FormInput name="email" />))
    const input = getByRole('textbox')
    fireEvent.change(input, { target: { value: 'novo valor' } })
    fireEvent.change(input, { target: { value: '' } })
    expect((input as HTMLInputElement).value).toBe('')
  })

  it('deve renderizar o label corretamente', () => {
    mockUseField('username')
    const { getByLabelText } = render(global.app(<FormInput name="username" label="Username" />))
    expect(getByLabelText('Username')).toBeInTheDocument()
  })

  it('deve alternar a visibilidade da senha ao clicar no botão', () => {
    mockUseField('password')
    const { getByLabelText, getByRole } = render(global.app(<FormInput name="password" type="password"
                                                                             label="Password" />))
    const input = getByLabelText('Password') as HTMLInputElement
    const toggleButton = getByRole('button')

    expect(input.type).toBe('password')

    fireEvent.click(toggleButton)
    expect(input.type).toBe('text')

    fireEvent.click(toggleButton)
    expect(input.type).toBe('password')
  })

  it('deve aplicar a função de formatação corretamente', () => {
    const formatMock = jest.fn((value: string) => value.toUpperCase())
    mockUseField('formattedInput')
    const { getByRole } = render(global.app(<FormInput name="formattedInput" format={formatMock} />))
    const input = getByRole('textbox')

    fireEvent.change(input, { target: { value: 'test' } })
    expect(formatMock).toHaveBeenCalledWith('test')
    expect((input as HTMLInputElement).value).toBe('TEST')
  })

  it('deve submeter e limpar o input do tipo telefone quando o comprimento for 15', async () => {
    mockUseField('phone')
    const mockRequestSubmit = jest.fn()
    const mockBlur = jest.fn()

    const originalRequestSubmit = window.HTMLFormElement.prototype.requestSubmit
    window.HTMLFormElement.prototype.requestSubmit = mockRequestSubmit

    const originalBlur = HTMLInputElement.prototype.blur
    HTMLInputElement.prototype.blur = mockBlur

    const { getByRole } = render(
      global.app(
        <form onSubmit={mockRequestSubmit}>
          <FormInput name="phone" type="tel" autoSubmitTelefone />
        </form>
      )
    )

    const input = getByRole('textbox') as HTMLInputElement

    fireEvent.change(input, { target: { value: '123456789012345' } })

    await waitFor(() => {
      expect(mockRequestSubmit).toHaveBeenCalled()
      expect(mockBlur).toHaveBeenCalled()
    })

    HTMLInputElement.prototype.blur = originalBlur
    window.HTMLFormElement.prototype.requestSubmit = originalRequestSubmit
  })

  it('deve exibir o ícone de erro quando houver erro', () => {
    mockUseField('email', '', 'Campo inválido')
    const { getByText, getByRole } = render(global.app(<FormInput name="email" />))
    expect(getByText('Campo inválido')).toBeInTheDocument()
    const input = getByRole('textbox')
    expect(input).toHaveStyle(`box-shadow: 0 0.25rem 0.25rem ${theme.colors.container.outline.default}`)
  })
})
