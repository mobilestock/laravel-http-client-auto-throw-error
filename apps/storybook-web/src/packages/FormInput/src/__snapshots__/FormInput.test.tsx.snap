// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component - Web deve corresponder ao snapshot e verificar o uso do campo 1`] = `
<DocumentFragment>
  <div
    class="sc-gtLWhw hnFqUS"
  >
    <div>
      <input
        id="email"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
